### ISSUE TITLE
[1.11.2] Suggestion: Allow imp spawning in non-Vanilla biomes

### ISSUE THREAD

**Issue by LachimHeigrim** [progwml6/Natura#390]
It seems Natura's imps are unable to spawn in non-Vanilla biomes. 

I suspect it's due to `getCanSpawnHere()` method checking `this.world.provider instanceof WorldProviderHell` in [ EntityImp.java ]( ../../Natura/tree/1.11/src/main/java/com/progwml6/natura/entities/entity/passive/EntityImp.java ). I've got Nitro Creepers spawning without problems and this check is not present in their implementation.

For full reference, in the modpack I'm setting `worldProvider` to BoP using Just Another Dimension as suggested in #352, then adding Imps to all Nether biomes using BiomeTweaker. The imps seem to be spawning in (vanilla) Hell biome, but not in any of BoP biomes (e.g. Visceral Heap, Fungi Forest etc.), although they are added to spawn lists - at least from what I can see in BiomeTweaker debug outputs.

Versions: 
- BiomesOPlenty-1.11.2-6.2.0.2267-universal.jar
- natura-1.11.2-4.2.0.24.jar
- BiomeTweaker-1.11.2 3.0.281
- justenoughdimensions-1.11.2-1.5.4.jar

**Comment by alexbegt** [comment]
The imp should be spawning in any biome that has the type of "Nether", It could be that BOP's do not haver that type?

**Comment by LachimHeigrim** [comment]
Good catch. Looking at dictionary types for the BoP Nether biomes, neither of them has NETHER type. Below is BiomeTweaker output for Undergarden for example. I try to work around it with BiomeTweaker, maybe I could copy it using `inheritProperties()`. If not I raise an issue with BoP.

```json 
{
  "ID": 107,
  "Name": "Undergarden",
  "Resource Location": "biomesoplenty:undergarden",
  "Class": "biomesoplenty.common.biome.nether.BiomeUndergarden",
  "Root Height": 0.1,
  "Height Variation": 0.2,
  "Top Block": "biomesoplenty:grass",
  "Top Block Meta": "6",
  "Filler Block": "minecraft:netherrack",
  "Filler Block Meta": "0",
  "Ocean Top Block": "Disabled. Activate in BiomeTweakerCore.",
  "Ocean Top Block Meta": "Disabled. Activate in BiomeTweakerCore.",
  "Ocean Filler Block": "Disabled. Activate in BiomeTweakerCore.",
  "Ocean Filler Block Meta": "Disabled. Activate in BiomeTweakerCore.",
  "Actual Filler Blocks": "Disabled. Activate in BiomeTweakerCore.",
  "Grass Color": "Disabled. Activate in BiomeTweakerCore.",
  "Foliage Color": "Disabled. Activate in BiomeTweakerCore.",
  "Water Color": "16777215",
  "Temperature": 2.0,
  "Humidity": 0.0,
  "Water Tint": 16777215,
  "Enable Rain": false,
  "Enable Snow": false,
  "Dictionary Types": [
    "HOT",
    "PLAINS",
    "DRY",
    "SAVANNA"
  ]
...<shortened>..
}
```
